nvflare/ha/ha_admin_cmds.py

```diff
@@ -70,60 +70,73 @@ def list_sp(self, args, ctx: CommandContext):
         """
         overseer_agent = self.overseer_agent
         return {
-            "status": APIStatus.SUCCESS,
-            "details": str(overseer_agent.overseer_info),
-            "data": overseer_agent.overseer_info,
+            ProtoKey.STATUS: APIStatus.SUCCESS,
+            ProtoKey.DETAILS: str(overseer_agent.overseer_info),
+            ProtoKey.DATA: overseer_agent.overseer_info,
         }
 
     def get_active_sp(self, args, ctx: CommandContext):
         overseer_agent = self.overseer_agent
-        return {"status": APIStatus.SUCCESS, "details": str(overseer_agent.get_primary_sp())}
+        sp = overseer_agent.get_primary_sp()
+        return {ProtoKey.STATUS: APIStatus.SUCCESS, ProtoKey.DETAILS: str(sp), ProtoKey.META: sp.__dict__}
 
     def promote_sp(self, args, ctx: CommandContext):
         overseer_agent = self.overseer_agent
         if len(args) != 2:
-            return {"status": APIStatus.ERROR_SYNTAX, "details": "usage: promote_sp example1.com:8002:8003"}
+            return {
+                ProtoKey.STATUS: APIStatus.ERROR_SYNTAX,
+                ProtoKey.DETAILS: "usage: promote_sp example1.com:8002:8003",
+                ProtoKey.META: {MetaKey.STATUS: MetaStatusValue.SYNTAX_ERROR},
+            }
 
         sp_end_point = args[1]
         resp = overseer_agent.promote_sp(sp_end_point)
-        if json.loads(resp.text).get("Error"):
+        err = json.loads(resp.text).get("Error")
+        if err:
             return {
-                "status": APIStatus.ERROR_RUNTIME,
-                "details": "Error: {}".format(json.loads(resp.text).get("Error")),
+                ProtoKey.STATUS: APIStatus.ERROR_RUNTIME,
+                ProtoKey.DETAILS: f"Error: {err}",
+                ProtoKey.META: {MetaKey.STATUS: MetaStatusValue.INTERNAL_ERROR, MetaKey.INFO: err},
             }
         else:
+            info = f"Promoted endpoint: {sp_end_point}. Synchronizing with overseer..."
             return {
-                "status": APIStatus.SUCCESS,
-                "details": "Promoted endpoint: {}. Synchronizing with overseer...".format(sp_end_point),
+                ProtoKey.STATUS: APIStatus.SUCCESS,
+                ProtoKey.DETAILS: info,
+                ProtoKey.META: {
+                    MetaKey.STATUS: MetaStatusValue.OK,
+                    MetaKey.INFO: info,
+                },
```

Comment:
should we also use "make_meta" method here?

Reply:
Yes

Reply:
done as suggested